tests/utils.py

```diff
@@ -56,20 +57,28 @@ def start_privileged_container(
     )
 
 
+def wait_for_log(container: Container, log: str, timeout: int = 60) -> None:
+    try:
+        wait_event(timeout, Event(), lambda: re.search(log.encode(), container.logs(), re.DOTALL) is not None)
+    except TimeoutError:
+        print(container.logs())
+        raise
+
```

Comment:
Not a good idea, if you want to `wait_for_log` twice for the same message, in the second time onward it will always return immediately.

This `wait_for_log` function needs to be stateful and always start from the offset where the previous call found the log

Reply:
My current use actually checks that the message appears multiple times to handle that. I agree that it makes more sense to follow the log and check from the last offset, yeah